src/components/ActionSheetAwareScrollView/index.ios.js

```diff
@@ -0,0 +1,527 @@
+import React, {
+    useContext, useEffect,
+    createContext, forwardRef, useCallback, useMemo,
+} from 'react';
+import PropTypes from 'prop-types';
+import {Dimensions, Keyboard, ScrollView} from 'react-native';
+import Reanimated, {
+    runOnUI,
+    useSharedValue,
+    useWorkletCallback,
+    Easing,
+    KeyboardState,
+    useAnimatedKeyboard,
+    useAnimatedStyle, useDerivedValue, withTiming, runOnJS,
+} from 'react-native-reanimated';
+import {makeRemote} from 'react-native-reanimated/src/reanimated2/core';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import getComponentDisplayName from '../../libs/getComponentDisplayName';
+import styles from '../../styles/styles';
+import Log from '../../libs/Log';
+
+// When you need to debug state machine change this to true
+const DEBUG_MODE = false;
+
+function log(message, params) {
```

Comment:
Can we please:

1. Extract this to a separate library like `src/libs/WorkletLog`?
1. Implement the other log levels as well:
    - `info`
    - `alert`
    - `warn`
    - `hmmm`
    - `client`

Reply:
I don't think it's worth it to even do it since you should never do "logging" in workelts except when debugging them. Since calling things from the JS thread on the UI thread is overhead you don't want to have in animations at all.

I left this since it's useful when adding new stuff. But it basically does nothing when "DEBUG_MODE" is off.

Reply:
It's either:

- worth it, in which case we should move it
- not worth it, in which case we should remove it

but we should not leave it here as-is

Reply:
I can move it inside the state machine hook since it's used only in there and is useful only there.
There is a plan to add this same support for Android, and removing it completely feels wrong.

Reply:
> What's the best place to put library-like hooks?

You can create a new directory at `src/hooks` for custom hooks.

Reply:
Moved this hook into its own directory/file